### ISSUE TITLE
Return compilation errors from compileToTofu

### ISSUE THREAD

**Issue by robfig** [google/closure-templates#53]
When developing templates, it would be helpful to get the template, filename, line number, and error message for any compilation errors.  That allows me to show the detailed error message along with the line of source code in context to the developer.  I do that today by parsing the SoySyntaxException's message, after extracting the original exception from the suppressed frames.  Luke suggested that this use case could be better supported.

**Comment by mikesamuel** [comment]
To clarify, you don't have a reference to the `SoySyntaxException`, so can't call `getSourceLocation()`.  Instead, you've got access to the stderr and would like that stderr to be better formatted.

If that's correct, then it sounds like this could be addressed by having the main method for whatever you're running do call [`Thread.currentThread().setUncaughtExceptionHandler(...)`](http://docs.oracle.com/javase/7/docs/api/java/lang/Thread.html#setUncaughtExceptionHandler)
with something that checks for instances of SoySyntaxException to extract that and delegating to the default uncaught exception handler.

This exception handler muckery could be packaged up in a utility class.

**Comment by robfig** [comment]
No, I do have the SoySyntaxException, but the actual error is suppressed. I can work around that, but was asked to file this feature request from this discussion: 
https://groups.google.com/d/msg/closure-templates-discuss/Zh1Vu2hm6gk/Oy_suwuCAQAJ
